### PR TITLE
warn on KillMode=none, inform about left-over processes on stop and warn about sysv services

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/sysv-generator/sysv-generator.c [hunk 1]
```diff
@@ -788,6 +788,10 @@ static int enumerate_sysv(const LookupPaths *lp, Hashmap *all_services) {
                         if (!fpath)
                                 return log_oom();
 
+                        log_warning("SysV service '%s' lacks a native systemd unit file. "
+                                    "Automatically generating a unit file for compatibility. "
+                                    "Please update package to include a native systemd unit file, in order to make it more safe and robust.", fpath);
```
Comment (keszybz) [inline src/sysv-generator/sysv-generator.c:793]: I'm not convinced that those warnings without any new lines will be readable. But maybe let's merge this, and adjust the format later if necessary.

**Comment by keszybz** [comment]
centos ci (arch in kvm) failed with a timeout in TEST-43-PRIVATEUSER-UNPRIV, looks unrelated.

**Comment by mbiebl** [comment]
Might be of interest here:
https://lintian.debian.org/tags/missing-systemd-service-for-init.d-script.html

There are around 1200 packages in the Debian archive, shipping 1300 SysV init scripts.
According to lintian, we have 380 packages which don't ship a native systemd service yet.

**Comment by mbiebl** [comment]
I.e., we already generate such a warning for the package maintainer, not the user

**Comment by mrc0mmand** [comment]
> centos ci (arch in kvm) failed with a timeout in TEST-43-PRIVATEUSER-UNPRIV, looks unrelated.

Should be, hopefully, fixed by #15923.

**Comment by poettering** [comment]
> Might be of interest here:
> https://lintian.debian.org/tags/missing-systemd-service-for-init.d-script.html
> 
> There are around 1200 packages in the Debian archive, shipping 1300 SysV init scripts.
> According to lintian, we have 380 packages which don't ship a native systemd service yet.

Ah, that's interesting to know.

> I.e., we already generate such a warning for the package maintainer, not the user

Yeah, having that in the packaging scripts makes a ton of sense, and actually more than having it in the generator. However, this is really actually faced to slowly put more pressure on 3rd party vendors, who still stick to sysvinit scripts. i.e. those weird commercial, super-slow-moving stuff. i.e. closed source databases, closed source storage stacks and the like. That kind of stuff isn't build inside of distros, hence the packaging scripts are not as effective.

I think this is a very weak way to increase the push for people to migrate their stuff, after all it's a log message that is easily ignored. systemd has been in the big distros for long enough, I think one additional explanatory message with a friendly tone should be fine to add...

**Comment by mbiebl** [comment]
> > Might be of interest here:
> > https://lintian.debian.org/tags/missing-systemd-service-for-init.d-script.html
> > There are around 1200 packages in the Debian archive, shipping 1300 SysV init scripts.
> > According to lintian, we have 380 packages which don't ship a native systemd service yet.

Might be a nice idea to add that to https://trends.debian.net/.
In case someone using Debian is reading this and wants to contribute something.

> > I.e., we already generate such a warning for the package maintainer, not the user
> 
> Yeah, having that in the packaging scripts makes a ton of sense, and actually more than having it in the generator. However, this is really actually faced to slowly put more pressure on 3rd party vendors, who still stick to sysvinit scripts. i.e. those weird commercial, super-slow-moving stuff. i.e. closed source databases, closed source storage stacks and the like. That kind of stuff isn't build inside of distros, hence the packaging scripts are not as effective.
> 
> I think this is a very weak way to increase the push for people to migrate their stuff, after all it's a log message that is easily ignored. systemd has been in the big distros for long enough, I think one additional explanatory message with a friendly tone should be fine to add...

This wasn't really meant as an objection. Just wanted to provide some additional data.

**Comment by halfline** [comment]
Note this changed showed up in the plymouth tracker today... https://gitlab.freedesktop.org/plymouth/plymouth/-/issues/123

plymouth relies on KillMode=none to weather the initramfs -> rootfs transition.

**Comment by poettering** [comment]
@halfline I commented there now.

**Comment by vincentbernat** [comment]
I think there are valid usages of `KillMode=None`. For example, it provides an easy way to handle process upgrade without impacting existing connections: it lets the old process to handle current long-running connections while systemd starts a new process. I am describing this usage [here](https://vincent.bernat.ch/en/blog/2018-systemd-golang-socket-activation#waiting-longer-for-existing-connections). I have an alternative, but it is far more complex.

**Comment by Werkov** [comment]
I second to @vincentbernat. My proposal is to _not_ drop `KillMode=none` until something like #10228 is implemented and such use cases can be adopted to it. (OTOH, `KillMode=none` has its own drawbacks so deprecation is justified in other use cases.)

**Pull request merged**